samples/apps/copilot-chat-app/webapi/Controllers/SemanticKernelController.cs

```diff
@@ -144,19 +144,21 @@ public class SemanticKernelController : ControllerBase, IDisposable
     /// </summary>
     private async Task RegisterPlannerSkillsAsync(CopilotChatPlanner planner, PlannerOptions options, OpenApiSkillsAuthHeaders openApiSkillsAuthHeaders, ContextVariables variables)
     {
-        // Register the Klarna shopping ChatGPT plugin with the planner's kernel.
-        using DefaultHttpRetryHandler retryHandler = new(new HttpRetryConfig(), this._logger)
-        {
-            InnerHandler = new HttpClientHandler() { CheckCertificateRevocationList = true }
-        };
-        using HttpClient importHttpClient = new HttpClient(retryHandler, false);
-        importHttpClient.DefaultRequestHeaders.Add("User-Agent", "Microsoft.CopilotChat");
-        await planner.Kernel.ImportChatGptPluginSkillFromUrlAsync("KlarnaShoppingSkill", new Uri("https://www.klarna.com/.well-known/ai-plugin.json"),
-            importHttpClient);
-
-        //
         // Register authenticated skills with the planner's kernel only if the request includes an auth header for the skill.
-        //
+
+        // Klarna Shopping
+        if (openApiSkillsAuthHeaders.KlarnaAuthentication != null)
+        {
+            // Register the Klarna shopping ChatGPT plugin with the planner's kernel.
+            using DefaultHttpRetryHandler retryHandler = new(new HttpRetryConfig(), this._logger)
+            {
+                InnerHandler = new HttpClientHandler() { CheckCertificateRevocationList = true }
+            };
+            using HttpClient importHttpClient = new HttpClient(retryHandler, false);
+            importHttpClient.DefaultRequestHeaders.Add("User-Agent", "Microsoft.CopilotChat");
```

Comment:
Why do we want to override this for Klarna specifically?

Reply:
We should be providing a proper HttpClient everywhere, but I'm starting here.